src/App.tsx

```diff
@@ -1,16 +1,143 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Error } from './types/Error';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  changeTodo,
+} from './api/todos';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoError } from './components/TodoError/TodoError';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [userTodos, setUserTodos] = useState<Todo[]>([]);
+  const [hasTodos, setHasTodos] = useState(false);
+  const [hasError, setHasError] = useState<Error>({ status: false });
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [changingIds, setChangingIds] = useState<number[]>([0]);
+
+  const handleSetError = useCallback((error: Error) => {
+    setHasError(error);
+    setTimeout(() => {
```

Comment:
not critical, but should we pass full error object here? I mean if we use this method while errors are appeared, so we'll always have `{ status: true }` part here, maybe let's pass only message here?